iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -621,170 +745,222 @@ private async Task RefreshSasTokenLoopAsync(DateTime refreshesOn, CancellationTo
 
         private async Task VerifyIsOpenAsync(CancellationToken cancellationToken)
         {
-            await _handlerSemaphore.WaitAsync(cancellationToken);
+            if (Logging.IsEnabled)
+                Logging.Enter(this, cancellationToken, nameof(VerifyIsOpenAsync));
 
             try
             {
-                if (!_isOpen)
+                ThrowIfDisposed();
+                await _clientOpenSemaphore.WaitAsync(cancellationToken);
+
+                try
                 {
-                    throw new InvalidOperationException($"The client connection must be opened before operations can begin. Call '{nameof(OpenAsync)}' and try again.");
+                    if (GetClientTransportStatus() != ClientTransportStatus.Open)
```

Comment:
This will result in operations invoked when the client is "opening" to also get rejected. It seemed like a reasonable expectation since we do want the client behavior to be deterministic.